qiskit/circuit/quantumcircuit.py

```diff
@@ -245,7 +256,80 @@ def __getitem__(self, item):
         """Return indexed operation."""
         return self.data[item]
 
+    @staticmethod
+    def _bit_argument_expansion(bit_representation, in_array):
```

Comment:
`expansion` might not be the right name here, since we're not doing the broadcasting/expansion which happens under `Instruction`. The length of the tuple returned here should always match the length of the `bit_representation`, yes? The goal here is that we want qargs after this point to be always be of a standardized type.  Maybe argument coercion, casting, conversion, ...?

Reply:
> The length of the tuple returned here should always match the length of the bit_representation, yes? 

No.. not always. For example, the following returns a list of the size of the Register `(bit_represeantion) == 1` :
```
            elif isinstance(bit_representation, Register):
                return bit_representation[:]
```

It is correct that the goal is to have bit tuples after that point.  I like `conversion`... what you think?

Reply:
Maybe "reduction"? There are what I call expansion is things like this `circuit.h(qr) -> circuit.h([qr[0], qr[1]])`

Reply:
:+1: for conversion (since not all of the cases will be expansions, e.g. `0 => qr[0]`). If we say expansion is e.g. `qr => [qr[0], qr[1]]`, should `argument_expansion` on `Instruction` be renamed to back to e.g. `broadcast_argument` or `argument_broadcast`? The exact names are less important, but it be nice to be clear on which transformations are expected to happen where.

Reply:
`argument_expansion -> argument_conversion` done in 432efcd2
`broadcast_arguments -> argument_expansion` done in 90d61f5b